### DISCUSSION TITLE
Backup USB/SD/MicroSD drive

### DISCUSSION THREAD

**Discussion by TheGates** [aonez/Keka#1019] [Q&A] [answered]
Hello,

I currently use dd to create an ISO on my RPi MicroSD card and then I run it through Keka as most of the MicroSD card is blank so I go from 128G down to 4G.  Is there a way to skip the dd part and have Keka create an archive that contains the entire partition structure of the MicroSD?  My goal is if I become a little too adventurous with Linux and I need to roll back that I can just restore the ISO to the MicroSD, put it back in the Pi, and boot back up to the date it was backed up.

Thanks!

**Comment by aonez** [comment]
This is currently not supported in Keka and due to sandboxing might be not possible ever...

**Comment by gingerbeardman** [comment, marked as answer]
I'd recommend using macOS _Disk Utility_ to do this. It can create a backup of a whole disk and you have the option to have the resulting file be _compresed_ (though as part of the disk image format, not as a zip). 

As I understand it _Disk Utility_ is really just a GUI for low level tools like `dd`, `diskutil`, etc. 

> https://support.apple.com/en-gb/guide/disk-utility/dskutl11888/mac
> Create a disk image using Disk Utility on Mac - Apple Support
> 
> See section: Create a disk image from a disk or connected device

If you want more fine-grained options for compressed disk images, like using LZFSE compression, try DropDMG (paid, free trial) https://c-command.com/dropdmg/
